orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/join/MinionAcceptor.java

```diff
@@ -65,6 +66,13 @@ private void removeMinionIdsInBothDeniedAndUnacceptedState(SaltConnector sc, Lis
             throw new CloudbreakOrchestratorFailedException("There were minions in denied and unaccepted state at the same time: " +
                     minionIdsInBothDeniedAndUnacceptedState);
         }
+
+        ArrayList<String> minionIdsOnlyDenied = new ArrayList<String>(deniedMinions);
+        minionIdsOnlyDenied.removeAll(unacceptedMinions);
+        if (!minionIdsOnlyDenied.isEmpty()) {
+            LOGGER.info("There are minions in denied state, removing: ", minionIdsOnlyDenied);
+            sc.wheel("key.delete", minionIdsOnlyDenied, Object.class);
+        }
```

Comment:
could you move this to a separate method, and I think this should be called outside of `removeMinionIdsInBothDeniedAndUnacceptedState`

I would create a method called `cleanupMinionIds` which would call this new method and `removeMinionIdsInBothDeniedAndUnacceptedState`